src/Utilities/Compiler/Extensions/IEnumerableExtensions.cs

```diff
@@ -103,6 +103,87 @@ public static void Dispose<T>(this IEnumerable<T> collection)
+        /// <returns><see langword="true" /> the source sequence contains, exactly, <paramref name="count"/> elements; otherwise, <see langword="false" />.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static bool HasExactly<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            using (var enumerator = source.GetEnumerator())
+            {
+                while (count-- > 0)
+                {
+                    if (!enumerator.MoveNext())
+                    {
+                        return false;
+                    }
+                }
+
+                return !enumerator.MoveNext();
+            }
+        }
+
+        /// <summary>
+        /// Determines whether a sequence contains more than <paramref name="count"/> elements.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <param name="source">The <see cref="IEnumerable{TSource}"/> to check for cardinality.</param>
+        /// <param name="count">The number of elements to ensure exists.</param>
+        /// <returns><see langword="true" /> the source sequence contains more than <paramref name="count"/> elements; otherwise, <see langword="false" />.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static bool HasMoreThan<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            using (var enumerator = source.GetEnumerator())
+            {
+                while (count-- > 0)
+                {
+                    if (!enumerator.MoveNext())
+                    {
+                        return false;
+                    }
+                }
+
+                return enumerator.MoveNext();
+            }
+        }
+
+        /// <summary>
+        /// Determines whether a sequence contains less than <paramref name="count"/> elements.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <param name="source">The <see cref="IEnumerable{TSource}"/> to check for cardinality.</param>
+        /// <param name="count">The number of elements to ensure exists.</param>
+        /// <returns><see langword="true" /> the source sequence contains less than <paramref name="count"/> elements; otherwise, <see langword="false" />.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static bool HasLessThan<TSource>(this IEnumerable<TSource> source, int count)
```

Comment:
Gramatically, `HasFewerThan` would be more correct (also `HasGreaterThan`) as `source` is countable, although usage of 'less' in such cases is slowly becoming more commonplace.